### PR TITLE
Add support for both scala nad java duration into timer

### PR DESCRIPTION
**Pull request by tomasherman**
It's very common to run into both scala and java duration in the Scala code. There is no need to push this onto users when it comes to metrics and it's trivial for us to support both. Also they are basically both parts of STD libraries so i think we should have support baked in

### CODE REVIEW

File: code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 import java.time.Duration
+import scala.concurrent.duration.FiniteDuration
 
 import com.avast.datadog4s.api.Tag
 
 trait Timer[F[_]] {
   def time[A](f: F[A], tags: Tag*): F[A]
 
-  def record(duration: Duration, tags: Tag*): F[Unit]
+  def record(duration: Duration, tags: Tag*): F[Unit]       = recordMillis(duration.toMillis, tags: _*)
```
Comment (hanny24) [inline code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala:11]: I'm not really a big fan of method overloading as it makes it much harder to pass method as value, ie. `timer.record _` can't really be used.

We could also make the method generic and require some sort of typeclass instance for that type. NOt sure if it's worth though.

Comment (tomasherman) [reply code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala]: i decided to add the typeclass, no need to limit it to just couple of certain types

File: code/api/src/main/scala/com/avast/datadog4s/api/metric/AsDuration.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.avast.datadog4s.api.metric
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.FiniteDuration
+
+trait AsDuration[A] { self =>
+  def valueOfTimeUnit(a: A, timeUnit: TimeUnit): Long
```
Comment (hanny24) [inline code/api/src/main/scala/com/avast/datadog4s/api/metric/AsDuration.scala:8]: Just a tip: `extractUnit`  seems like a better name to me.

File: code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
-  def record(duration: Duration, tags: Tag*): F[Unit]
+  def record(duration: Duration, tags: Tag*): F[Unit] = recordT[Duration](duration, tags: _*)
+
+  /**
+   * Record raw measurement represented as long
+   * @param duration Measured duration
+   * @param timeUnit TimeUnit in which duration is measured
+   * @param tags Tags that should be applied for this recording
+   */
+  def recordRaw(duration: Long, timeUnit: TimeUnit, tags: Tag*): F[Unit] =
```
Comment (hanny24) [inline code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala:19]: `recordWithUnit` ?

File: code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
 import com.avast.datadog4s.api.Tag
 
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
 trait Timer[F[_]] {
   def time[A](f: F[A], tags: Tag*): F[A]
 
-  def record(duration: Duration, tags: Tag*): F[Unit]
+  def record(duration: Duration, tags: Tag*): F[Unit] = recordT[Duration](duration, tags: _*)
```
Comment (hanny24) [inline code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala:11]: Do we really want to keep this? We are going to introduce binary incompatibility anyway. We can keep only `def record[T](value: T ...` which should be source compatible for the most of times.

Comment (tomasherman) [reply code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala:11]: good point

File: code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala (hunk 3 above)
Comment (tomasherman) [reply code/api/src/main/scala/com/avast/datadog4s/api/metric/Timer.scala]: fixed

**Review by hanny24 — APPROVED** [review]

**Pull request merged**